src/cplus_plugin/utils.py

```diff
@@ -1477,3 +1478,68 @@ def create_connectivity_raster(
         logs.append(traceback.format_exc())
 
     return False, logs
+
+
+def normalize_raster(
+    input_raster_path: str,
+    output_raster_path: str,
+    processing_context: QgsProcessingContext = None,
+    feedback: QgsProcessingFeedback = None,
+):
+    """
+    Create a normalized input raster
+
+    :param input_raster_path: Input layer path
+    :type input_raster_path: str
+
+    :param output_raster_path: Output layer path
+    :type output_raster_path: str
+
+    :param processing_context: Qgis processing context
+    :type processing_context: QgsProcessingContext, default None
+
+    :param feedback: Qgis processing feedback
+    :type feedback: QgsProcessingFeedback
+    """
+    try:
+        input_raster_layer = QgsRasterLayer(input_raster_path, "Input Raster")
+
+        if not input_raster_layer.isValid():
+            return False, f"Invalid raster layer {input_raster_path}"
+
+        provider = input_raster_layer.dataProvider()
+        band_statistics = provider.bandStatistics(1)
+        min_value = band_statistics.minimumValue
+        max_value = band_statistics.maximumValue
+
+        if min_value is None or max_value is None:
+            return False, f"Raster layer has no valid statistics, {input_raster_path}"
+
+        if min_value >= 0 and max_value <= 1:
+            return (
+                True,
+                f"Layer is already normalized (min={min_value}, max={max_value})",
+            )
+
+        expression = f"(A - {min_value}) / ({max_value} - {min_value})"
```

Comment:
If all pixels are 0, this could result in error. Unless it has been validated somewhere?